scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -629,29 +630,35 @@ def apply_filters(self, **kwargs):
         emu_filter = self.options.emulation_only
 
         logger.debug("platform filter: " + str(platform_filter))
+        logger.debug("  vendor filter: " + str(vendor_filter))
         logger.debug("    arch_filter: " + str(arch_filter))
         logger.debug("     tag_filter: " + str(tag_filter))
         logger.debug("    exclude_tag: " + str(exclude_tag))
 
         default_platforms = False
+        vendor_platforms = False
         emulation_platforms = False
 
         if all_filter:
             logger.info("Selecting all possible platforms per test case")
             # When --all used, any --platform arguments ignored
             platform_filter = []
-        elif not platform_filter and not emu_filter:
+        elif not platform_filter and not emu_filter and not vendor_filter:
             logger.info("Selecting default platforms per test case")
             default_platforms = True
         elif emu_filter:
             logger.info("Selecting emulation platforms per test case")
             emulation_platforms = True
+        elif vendor_filter:
+            vendor_platforms = True
```

Comment:
Please add

`logger.info(f"Selecting platforms from {vendor_filter} vendor(s) per test case")`

to keep the same level of information as in the above cases.